python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -156,10 +156,10 @@ def evaluate(
                 label_col=self.feature_config.get_label("node_name"),
                 old_rank_col=self.feature_config.get_rank("node_name"),
                 new_rank_col=RankingConstants.NEW_RANK,
-                group_keys=self.feature_config.get_group_metrics_keys(
-                    "node_name"),
-                secondary_labels=self.feature_config.get_secondary_labels(
-                    "node_name"),
+                group_keys=list(set(self.feature_config.get_group_metrics_keys(
```

Comment:
is it expected to have duplicates and you are taking the `set`? is it a sign of an error?

Reply:
I think I know what the issue is. I can try and fix it, but it will be messy. I think we should keep it as a separate PR. I will file a bug for this if that works for you. This small change suppresses a lot of pandas warnings for now.

I can also remove this change from this PR and keep it in that bug itself, if that's better.

Reply:
It works. I let you decide what you believe the best approach is. If you go with keeping it, just make a note to the bug to simplify these lines once we solve it.